_posts/2018-09-03-rest-review.md

```diff
@@ -0,0 +1,99 @@
+---
+author: Jorge Gueorguiev Garcia
+layout: post
+asset-type: post
+title: "Rest Review"
+date: 2018-08-30 08:01:00
+description: Where I talk about what is, and what is not, the REST architecture.
+image: 
+    src: /assets/custom/img/blog/2018-08-13-automated-code-styling/caffeine-coffe-cup-6347.jpg
+tags: 
+- REST
+- Architecture
+---
+
+There are two events that have happened recently close to each other. The first one was a decision to actually learn GraphQl. I have been thinking about it for a while, but now is time to actually put the effort on it. The second one was a talk with my colleague [Luciano](https://codurance.com/publications/author/luciano-palma/) in which I incorrectly made a remark about resources on REST. Because of both (for the former as I want to compare, for the latter to assert my knowledge), I decided to re-read both the [Fielding PhD dissertation](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm) and the book [Rest in Practice](http://restinpractice.com/book/) by Robinson, Webber and Parastatidis.
+
+So here is a review of REST.
+
+## The Sources
+
+The sources (plus the actual practice) forms the basis of my understanding of REST. Do read what I say here, but do go back to them, read them yourself and check if what I write makes sense.
+
+### Fielding's dissertation
+
+The first three chapters about analysis of (web) architectures are fantastic. He offers his own classification of network architectures and how they built on top of each other. It creates an easy to follow taxonomy and shows how he arrived at REST. I found quite interesting as well that for him the architecture is, basically, the snapshot of the running system.
+
+One thing that I completely forgot is that he worked on quite a few projects like Http/1.0, Http/1.1, Apache, ... and that Http itself is based on REST (mostly, as he does talk about the definitely not RESTful cookies and the need to support non RESTful existing extensions).
+
+### Rest In Practice
+
+I think this was a fantastic book with a really good introduction to REST apis. When it came the time for me to work on my first Web Api, I read this book as it had all glowy recomendations. After reading it a second time a few years later, I can say that is still fresh and on point.
+
+## Web Api != REST Api
+
+Having an API that uses http doesn't mean that you are have a REST Api. I think this is an important distinction, as a lot of programmers I have seen do mix them. A Web API is an API that can be accessed trought the network. A REST api is an api that follows the REST architectural style.
```

Comment:
Do you mean "can be accessed through the network"?

Reply:
Fixed